### DISCUSSION TITLE
Libre 2+ sensor 15 day life not recognised

### DISCUSSION THREAD

**Discussion by PhilTU** [NightscoutFoundation/xDrip#3833] [Bug reports]
xDrip+ was adamant my Libre 2+ sensor would end yesterday at 17:47. It's still running today at 20:49

xDrip says Age 15.1 days and no longer offers me an end time.

First world problem, but I know some stress about these issues.

**Comment by Navid200** [comment]
Can you show screenshots?

**Comment by PhilTU** [comment]
![Screenshot_20241227-211748](https://github.com/user-attachments/assets/1be83c13-d71c-4a82-8907-407bd2ad55c1)
This shows current run time with missing expiry

**Reply by Navid200** [reply]
And what does the classic status page identifies it as?

**Reply by PhilTU** [reply]
Serial number mainly correct, sensor start yesterday, Data source Limitter

Sensor start in xDrip is meaningless but in this case is from 26/12 which is when I re-connected this sensor to xDrip+

**Reply by PhilTU** [reply]
![image](https://github.com/user-attachments/assets/cf54be92-4247-4d6b-8644-9dc2dcbe33b6)

**Reply by Navid200** [reply]
Is this sensor something that required a change to xDrip to work in xDrip?
Or is it something that worked using an existing setup for a different sensor that works only 14 days?  And it worked without any updates to the xDrip code?

If the former? can you tell me when approximately the change was made to xDrip?

**Reply by PhilTU** [reply]
in late May 2024 the ability to use Libre 2+ sensors was added. They have an extra day life. 31/5/24 it was working for sure, 10 days prior to that was being worked on. Link to c6 sensor work https://github.com/NightscoutFoundation/xDrip/discussions/3439#discussioncomment-9611227 

Another user recently reported the incorrect life expectancy here too - https://github.com/NightscoutFoundation/xDrip/discussions/3631#discussioncomment-10711789

**Reply by Navid200** [reply]
How could it have been working on May 24 considering you comment on May 31 is your first confirmation that it works and you got the test files from Tzachi a day before that?  Can you clarify?
https://github.com/NightscoutFoundation/xDrip/discussions/3439#discussioncomment-9613945

**Reply by PhilTU** [reply]
> How could it have been working on May 24

I didn't say that.

First update on 20/5/24 didn't work for UK L2+ https://github.com/NightscoutFoundation/xDrip/discussions/3439#discussioncomment-9498943

**Reply by Navid200** [reply]
Then, I misunderstood this:  
![Screenshot 2024-12-30 083350](https://github.com/user-attachments/assets/b0e1de06-9b86-44f9-89df-9985488bcfa6)  
<br/>  
  
Is this when it became available for everyone?  
  
![Screenshot 2024-12-30 083603](https://github.com/user-attachments/assets/23535368-5158-444f-81b5-5c90799c6a99)

**Reply by PhilTU** [reply]
> Is this when it became available for everyone?

probably, I tested a test version from drive on 31/5 as the previous 20/5 attempt didn't solve it

**Reply by PhilTU** [reply]
I think you took "late May 2024" to mean "May 24th"

**Reply by Navid200** [reply]
Please convert to issue.  Thanks

**Comment by PhilTU** [comment]
This from a couple of days ago shows expiry and 12.7 day age at the time 
![Uploading Screenshot_20241225-095145.png…]()

**Comment by Navid200** [comment]
Are you currently using this type of sensor?  If I create a test version, will you be able to test?
I have no idea I even can.  But, it will be pointless if it cannot be tested.

**Reply by PhilTU** [reply]
I can test next week probably if you are able to do this, yes.

**Reply by sebmatrix86** [reply]
Hello @Navid200 , i can test now. I have this type of sensor on my arm with the S/N beginning with 301.
